numpy/_core/_methods.py

```diff
@@ -202,34 +191,25 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False, *,
     else:
         x = um.multiply(x, um.conjugate(x), out=x).real
 
-    ret = umr_sum(x, axis, dtype, out, keepdims=keepdims, where=where)
+    ret = umr_sum(
+        x, axis, dtype, Ellipsis if out is None else out,
+        keepdims=keepdims, where=where)
 
     # Compute degrees of freedom and make sure it is not negative.
     rcount = um.maximum(rcount - ddof, 0)
 
     # divide by degrees of freedom
-    if isinstance(ret, mu.ndarray):
-        ret = um.true_divide(
-                ret, rcount, out=ret, casting='unsafe', subok=False)
-    elif hasattr(ret, 'dtype'):
-        ret = ret.dtype.type(ret / rcount)
-    else:
-        ret = ret / rcount
-
+    # TODO: dtype=ret.dtype castst first and isn't quite ideal that way.
+    ret = um.true_divide(
+            ret, rcount, out=out, casting='unsafe', dtype=type(ret.dtype))
```

Comment:
As above, why not `out=ret`?

Reply:
This shows one of the trickier things... If we want to keep allowing a scalar return here (which we must), then `out=ret` in the last step doesn't work always.  Although, you could use it in some cases.

The solution is maybe really best as `conv = array_converter(...)` and then use `conv.wrap()` at the end, although it'll change the wrapping slightly (i.e. _not_ happening inside ufuncs)...  And that might break `np.ma.` or so...

Reply:
Yeah, I did this to be more sure it was correct(er) for starters.  I had some test failures related to `subclass[()]` going to scalar but the ufunc path (array-wrap) ignoring the `return_scalar=True`.